### PR TITLE
Run buildifier fix, clean up unused variables/functions

### PR DESCRIPTION
**Pull request by dierksen**
Too many squiggles when I'm working on stuff, so took a quick cleanup pass.

I'd love to figure out how to suppress errors on the `xcspecs` file and add a bunch more docstrings, but this is a baby step 😄

### CODE REVIEW

File: rules/library.bzl [hunk 1]
```diff
@@ -126,12 +126,10 @@ module {module_name} {{
 
 def _write_umbrella_header(
         name,
-        library_tools,
         generate_default_umbrella_header,
         public_headers = [],
-        private_headers = [],
         module_name = None,
-        **kwargs):
+        **_kwargs):
```
Comment (jerrymarino) [inline rules/library.bzl:132]: Remove?

File: rules/library.bzl [hunk 2]
```diff
@@ -177,7 +175,7 @@ FOUNDATION_EXPORT const unsigned char {module_name}VersionString[];
     )
     return destination
 
-def _generate_resource_bundles(name, library_tools, module_name, resource_bundles, platforms, **kwargs):
+def _generate_resource_bundles(name, library_tools, resource_bundles, platforms, **_kwargs):
```
Comment (jerrymarino) [inline rules/library.bzl:178]: Remove _kwargs?

File: rules/library.bzl [hunk 3]
```diff
@@ -194,7 +192,7 @@ def _generate_resource_bundles(name, library_tools, module_name, resource_bundle
         bundle_target_names.append(target_name)
     return bundle_target_names
 
-def _error_on_default_xcconfig(name, library_tools, default_xcconfig_name, **kwargs):
+def _error_on_default_xcconfig(name, default_xcconfig_name, **_kwargs):
```
Comment (jerrymarino) [inline rules/library.bzl:195]: Remove _kwargs?

**Review by jerrymarino — APPROVED** [review]
Thanks for the cleanup 🙇 - I wonder why the CI isn't picking up on this.. Left a few inline to drop unused args

**Comment by jerrymarino** [comment]
Also the `DEBUG` stuff in rules_ios is another one, I'm getting around this by filtering out this level of diags BTW:
```
build --ui_event_filters=-debug
```

File: rules/framework.bzl [hunk 4]
```diff
@@ -1021,7 +1016,7 @@ that this target depends on.
         ),
         "_headermap_builder": attr.label(
             executable = True,
-            cfg = "host",
+            cfg = "exec",
```
Comment (thiagohmcruz) [inline rules/framework.bzl:1019]: Was curious to know the implications of this change but apparently `host` was deprecated and effectively is pretty much the same as `exec` now. For other reviewers / future reference: https://bazel.build/extending/rules#configurations

**Review by thiagohmcruz — APPROVED** [review]
Thx for doing this!

File: rules/framework.bzl (hunk 4 above)
Comment (dierksen) [reply rules/framework.bzl:1019]: Yes, this is one that got handled by the autofix; most of the rest were semi-manual.

**Comment by jerrymarino** [comment]
@dierksen any chance you're using a newer buildifier fix than CI to catch this? If so we should probably upgrade to the more improved one to catch dead code or update settings

**Comment by dierksen** [comment]
Doing another pass with the unused `kwargs` cleaned up. Also bumped the buildifier version; looks like they may have tweaked a few things since 5.0.1.

**Review by luispadron — APPROVED** [review]
🧼

**Pull request merged**